ui/client/common/ProcessStateUtils.js

```diff
@@ -1,39 +1,134 @@
 import _ from "lodash"
 
+import {BACKEND_STATIC_URL} from "../config"
+import iconDeployed from "../assets/img/states/deployed.svg"
+import iconNotDeployed from "../assets/img/states/not-deployed.svg"
+import iconUnknown from "../assets/img/states/state-unknown.svg"
+import iconErrorState from "../assets/img/states/state-error.svg"
+import iconCanceled from "../assets/img/states/canceled.svg"
+import urljoin from "url-join"
+
+const localIconPatternRegexp = /^((http|https|ftp):\/\/)/
+
 export default {
-  ACTION_DEPLOY: "DEPLOY",
-  ACTION_CANCEL: "CANCEL",
+  STATUSES: {
+    UNKNOWN: "UNKNOWN",
+    NOT_DEPLOYED: "NOT_DEPLOYED",
+    RUNNING: "RUNNING",
+    CANCELED: "CANCELED",
+    ERROR: "ERROR"
+  },
+
+  ACTIONS: {
+    NOT_DEPLOYED: null,
+    DEPLOY: "DEPLOY",
+    CANCEL: "CANCEL"
+  },
+
+  DEFAULT_STATUS_TOOLTIPS: {
+    UNKNOWN: "Unknown state of the process..",
+    NOT_DEPLOYED: "The process has never been deployed.",
+    RUNNING: "The process has been successfully deployed.",
+    CANCELED: "The process has been successfully cancelled.",
+    ERROR: "There are some errors with state of process..",
+  },
+
+  DEFAULT_STATUS_ICONS: {
+    UNKNOWN: iconUnknown,
+    NOT_DEPLOYED: iconNotDeployed,
+    RUNNING: iconDeployed,
+    CANCELED: iconCanceled,
+    ERROR: iconErrorState
+  },
+
+  isRunning(state) {
+    return this.getStateStatus(state) === this.STATUSES.RUNNING
+  },
 
   isDeployed(process) {
-    const action = _.get(process, "deployment.action")
-    return action != null && action.toUpperCase() === this.ACTION_DEPLOY
+    return this.getProcessAction(process) === this.ACTIONS.DEPLOY
   },
 
-  isCanceled(deployment) {
-    const action = _.get(deployment, "deployment.action")
-    return action != null && action.toUpperCase() === this.ACTION_CANCEL
+  getStatusTooltip(status) {
+    return _.get(this.DEFAULT_STATUS_TOOLTIPS, status, this.DEFAULT_STATUS_TOOLTIPS[this.STATUSES.UNKNOWN])
   },
 
-  //TODO: display e.g. warnings, use it in Visualization (left panel)
-  getStatusClass(processState, shouldRun, statusesLoaded) {
-    const isRunning = _.get(processState, "isRunning")
-    if (isRunning) {
-      return "status-running"
-    } else if (shouldRun) {
-      return statusesLoaded ? "status-notrunning" : "status-unknown"
+  getStatusIcon(status) {
+    return _.get(this.DEFAULT_STATUS_ICONS, status, this.DEFAULT_STATUS_ICONS[this.STATUSES.UNKNOWN])
+  },
+
+  getStateStatus(state) {
+    const status = _.get(state, "status", null)
+
+    if (_.isUndefined(state) || (state && status === null)) {
+      return this.STATUSES.UNKNOWN
     }
-    return null
+
+    //TODO: In future it will be nice when API return state with status CANCELED
```

Comment:
Please check if it can be on backend site - let's keep this walkaround as close to source of problem as it possible. Or at least write more verbose comment, that in situation when process was ran and then was stopped api returns null - it is the only possible situation when there is null in response so we assume that null mean CANCELED

Reply:
I've added support for CANCELED / DURING CANCEL status.